server/src/main/java/org/opensearch/action/support/RetryableAction.java

```diff
@@ -162,6 +210,10 @@ public void onResponse(Response response) {
             }
         }
 
+        public abstract RetryingListener getRetryingListenerForNextRetry();
+
+        public abstract long getRetryDelay();
+
```

Comment:
We just need two extension methods right?
1. delayMillis
2. nextDelayMillisBound

Can we avoid a RetryingListener instance altogether?

Reply:
For delay we need single extension point only, which will return the current delay. We calculate the next delay and create the new RetryingListener with it and pass it to Runnable.

> Can we avoid a getRetryingListenerForNextRetry altogether?

The way RetryingListener gets initialize for both cases is different, for one we need to pass long dealyMillis and for other we need to pass the iterator. That's why we need to keep it.